src/aaz_dev/cli/controller/portal_cli_generator.py

```diff
@@ -0,0 +1,359 @@
+            cmd_info['confirmation'] = True
+
+    def fill_cmd_help(self, cmd_info, cmd_cfg, leaf):
+        help_info = {}
+        self.generate_cmd_learn_more(help_info, leaf)
+        option_list = []
+        for arg_group in cmd_cfg.arg_groups:
+            for arg in arg_group.args:
+                assert arg.options is not None
+                if len(arg.options) > 0:
+                    # parameter set ignore 1 character option
+                    option_list += filter(lambda opt: len(opt) > 1, arg.options)
+        parameters_list = [ "[--" + param_name + "]" for param_name in option_list]
+        help_info['parameterSets'] = [{'parameters': parameters_list}]
+        cmd_info['help'] = help_info
+
+    def generate_default_example(self, cmd_info, leaf, var_option_list):
+        parameters = []
+        for (para_var, raw_options, is_required, is_in_path) in var_option_list:
+            if not is_required:
+                continue
+            pick_options = list(filter(lambda opt: len(opt) > 1, raw_options))
+            pick_option = pick_options[0]
+            parameters.append({
+                'name': ("-" if len(pick_option) == 1 else "--") + pick_option,
+                'value': "[" + para_var.strip("$") + "]"
+            })
+        if len(parameters) > 0:
+            cmd_info['examples'] = [{
+                'description': leaf.help.to_primitive()['short'],
+                'parameters': parameters
+            }]
+
+    def find_para_var(self, para, var_option_list):
+        for (var, options, is_required, is_in_path) in var_option_list:
+            if para in options:
+                return (var, options, is_required, is_in_path)
+        return (None, None, None, None)
+
+    def fill_cmd_examples(self, cmd_info, cmd_cfg, leaf, target_version):
+        var_option_list = []
+        var_required = set()
+        subId_required = False
+        for arg_group in cmd_cfg.arg_groups:
+            for arg in arg_group.args:
+                assert arg.options is not None
+                if len(arg.options) == 0: continue
+                is_required = arg.required
+                var_name = arg.var.replace("$Path", "$path")
+                if arg.required and var_name.find("subscriptionId") == -1:
+                    # for user input cmd example, ignore subscriptionId
+                    var_required.add(var_name)
+                if arg.required and var_name == "$path.subscriptionId":
+                    subId_required = True
+                if cmd_info['path'].find("{resourceGroupName}") != -1 and var_name == "$path.resourceGroupName":
+                    var_required.add(var_name)
+                    is_required = True
+                var_option_list.append((var_name, arg.options,
+                                        is_required,
+                                        var_name.find('$path') != -1))
+
+        if target_version.examples is None or len(target_version.examples) == 0:
```

Comment:
This can be simplified like
```suggestion
        if not target_version.examples:
```